rust/otel-arrow-rust/src/otlp/attributes/decoder.rs

```diff
@@ -62,3 +76,532 @@ where
         }
     }
 }
+
+/// Decodes the parent IDs from their transport optimized encoding to the actual ID values.
+///
+/// In the transport optimized encoding, the record batch is sorted by value Type, then
+/// attribute key, then attribute value. If subsequent rows have the same key & value, the
+/// parent IDs are delta encoded.
+///
+/// There are additional exceptions that end the sequence of delta encoding:
+/// - null values (we don't consider null values equal, even if current row & previous
+///   row value are both null)
+/// - value types Empty, Slice & Map are never considered equal
```

Comment:
I don’t remember exactly why I didn’t consider null values as equivalent to each other (same goes for empty values). It might be helpful to add a TODO somewhere to reevaluate the impact of this on a representative stream. But for now, we’ll keep the original logic.

Reply:
Sounds good. @lquerel I made an issue to track this https://github.com/open-telemetry/otel-arrow/issues/463 and updated comments in the code at the relevant locations to reference this issue